### PR TITLE
feat: Table Input for tools metadata (tool name and description) in component as tools 

### PR DESCRIPTION
**Pull request by edwinjosechittilappilly**
Merge to main only after https://github.com/langflow-ai/langflow/pull/4946

This pull request includes several changes to the backend components and workflows. The most important changes focus on enhancing the component tools with metadata handling, and updating various component definitions.

### Component Tools Enhancements:
* Updated `ComponentToolkit` class in `src/backend/base/langflow/base/tools/component_tool.py` to include metadata handling for tools. Added a new method `update_tools_metadata` to update tool names and descriptions based on provided metadata. [[1]](diffhunk://#diff-e6b607a8c5e404d2d308314cf694b33701ebcd21f8d564b6932c522d3a5e02cbL163-R165) [[2]](diffhunk://#diff-e6b607a8c5e404d2d308314cf694b33701ebcd21f8d564b6932c522d3a5e02cbR245-R259)
* Introduced a new constant `TOOLS_METADATA_INPUT_NAME` in `src/backend/base/langflow/base/tools/constants.py`.

### Component Definitions Updates:
* Added an `icon` attribute to the `SQLDatabaseComponent` class in `src/backend/base/langflow/components/langchain_utilities/sql_database.py`.
* Replaced `FloatInput` with `SliderInput` for the `temperature` field in `OpenAIModelComponent` class in `src/backend/base/langflow/components/models/openai.py`.

### Custom Component Enhancements:
* Updated `run_and_validate_update_outputs` method in `src/backend/base/langflow/custom/custom_component/component.py` to handle tool metadata input.
* Modified `to_toolkit` method to update tools metadata if available. Added a new method `_build_tools_metadata_input` to build tool metadata input. [[1]](diffhunk://#diff-22fd745de33f20981c7cbcbf3c91d4a6f8719de20761eb22cf42f8ed6ec50169L969-R982) [[2]](diffhunk://#diff-22fd745de33f20981c7cbcbf3c91d4a6f8719de20761eb22cf42f8ed6ec50169R1153-R1188)

### CODE REVIEW

**Comment by edwinjosechittilappilly** [comment]
Updating unit tests. According to the current changes!

File: src/backend/base/langflow/custom/custom_component/component.py [hunk 1]
```diff
@@ -1137,6 +1145,43 @@ def send_error(
...
         return Output(name=TOOL_OUTPUT_NAME, display_name=TOOL_OUTPUT_DISPLAY_NAME, method="to_toolkit", types=["Tool"])
+
+    def _build_tools_metadata_input(self):
+        tools = self.to_toolkit()
+        table_schema = [
+            {
+                "name": "name",
+                "display_name": "Name",
+                "type": "str",
+                "description": "Specify the name of the output field.",
```
Comment (anovazzi1) [inline src/backend/base/langflow/custom/custom_component/component.py:1161]: what do you thing if we create consts for those hardCoded values in description?

File: src/backend/base/langflow/custom/custom_component/component.py [hunk 2]
```diff
@@ -1137,6 +1145,43 @@ def send_error(
...
+            },
+            {
+                "name": "description",
+                "display_name": "Description",
+                "type": "str",
+                "description": "Describe the purpose of the output field.",
```
Comment (anovazzi1) [inline src/backend/base/langflow/custom/custom_component/component.py:1167]: what do you thing if we create consts for those hardCoded values in description?

**Review by anovazzi1 — COMMENTED** [review]
looks really nice, just small suggestions

**Comment by edwinjosechittilappilly** [comment]
https://github.com/user-attachments/assets/9cf9ed36-dea3-495f-86c5-603003de371f

File: src/backend/base/langflow/custom/custom_component/component.py (hunk 1 above)
Comment (edwinjosechittilappilly) [reply src/backend/base/langflow/custom/custom_component/component.py:1161]: Thats a Good Thought!
How about loading the Table schema list of Dict as a constant?

**Review by anovazzi1 — APPROVED** [review]
lgtm

**Pull request merged**